libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -440,19 +456,74 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
         if (success == false) {
             break;
         }
-        uint32_t* count_ptr = (uint32_t*) args;
-        (*count_ptr)++;
         if (rx_dynamic_transforms_topic.transforms_size > 0) {
-            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received tf2_msgs/TFMessage of length: %u",
+            debug("Received tf2_msgs/TFMessage of length: %u",
                           static_cast<unsigned>(rx_dynamic_transforms_topic.transforms_size));
         } else {
-            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received tf2_msgs/TFMessage: Insufficient size ");
+            debug("Received tf2_msgs/TFMessage: Insufficient size ");
         }
     }
     }
 
 }
 
+#include "AP_DDS_Service_Table.h"
+
+/*
+  callback on request completion
+ */
+void AP_DDS_Client::on_request_trampoline(uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, SampleIdentity* sample_id, ucdrBuffer* ub, uint16_t length, void* args)
+{
+    AP_DDS_Client *dds = (AP_DDS_Client *)args;
+    dds->on_request(uxr_session, object_id, request_id, sample_id, ub, length);
+}
+
+void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, uint16_t request_id, SampleIdentity* sample_id, ucdrBuffer* ub, uint16_t length)
+{
+    (void) request_id; (void) length;
+    switch (object_id.id) {
+    case services[to_underlying(ServiceIndex::ARMING_MOTORS)].rep_id: {
+        bool arm;
+        bool result;
+        const bool deserialize_success = ucdr_deserialize_bool(ub,&arm);
+        if (deserialize_success == false) {
+            break;
+        }
+
+        if (arm == true) {
+            debug("Request for arming recieved");
+            result = AP::arming().arm(AP_Arming::Method::DDS);
+        } else {
+            debug("Request for disarming recieved");
+            result = AP::arming().disarm(AP_Arming::Method::DDS);
+        }
+
+        const uxrObjectId replier_id = {
+            .id = services[to_underlying(ServiceIndex::ARMING_MOTORS)].rep_id,
+            .type = UXR_REPLIER_ID
+        };
+        uint8_t reply_buffer[8] {};
+        ucdrBuffer reply_ub;
+
+        ucdr_init_buffer(&reply_ub, reply_buffer, sizeof(reply_buffer));
+        const bool serialize_success = ucdr_serialize_bool(&reply_ub,result);
+        if (serialize_success == false) {
+            break;
+        }
+
+        request_sample_count++;
+
+        uxr_buffer_reply(uxr_session, reliable_out, replier_id, sample_id, reply_buffer, sizeof(result));
```

Comment:
should sizeof(result) be reply_ub.offset?
maybe use ucdr_buffer_length(reply_buffer) ?